core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -262,6 +262,10 @@ public InputStream getResourceAsStream(String nm) {
             throw new ClassNotFoundException(name);
         }
         try {
+            final String pkgName = getPackageNameFromClassName(name);
+            if (getPackage(pkgName) == null) {
+                definePackage(pkgName, null, null, null, null, null, null, null);
```

Comment:
This is subject to a race condition where two classes being defined in the same package by different threads can cause the second one to fail with an exception if the sequence is just right.  The fix would be to protect this with a block like:

```java
if (getPackage(pkgName) == null) {
    synchronized (getClassLoadingLock(pkgName)) {
        if (getPackage(pkgName) == null) {
            definePackage(pkgName, null, null, null, null, null, null, null); // also we'll want to get this from the manifest but we can worry about that later
        }
    }
}

Reply:
I was assuming that there would be some race, but didn't know of `getClassLoadingLock` solution. Thanks!!!